Magnet.cpp

```diff
@@ -0,0 +1,60 @@
+#include "Magnet.h"
+
+Magnet::Magnet(int height, int width)
+{
+    std::random_device rd;
+    std::mt19937 gen(rd());        //Using Mersenne twister
+    std::uniform_int_distribution<> distribution(0,1);    //Evenly 0s and 1s
+
+    spins.reserve(height);
+    for(int i=0;i<height;i++)
+    {
+        spins.push_back(std::vector<int>());    //Puts an empty vector on it
+        spins[i].reserve(width);
+        for(int j=0;j<width;j++)
+            spins[i].push_back(distribution(gen));
+    }
+}
+
+bool Magnet::update(int i,int j)
+{
+    if(updatePending(i,j))
+    {
+        spins[i][j] = spins[i][j] == 1 ? -1 : 1;
```

Comment:
What are the possible values of spins[i][j]? I believed it to just be {1,-1} always.

Reply:
You are correct, spins[i][j] has to be either 1 or -1 for any i, j.

Reply:
Then isn't this code is disgusting? What is wrong with spins[i][j] = -spins[i][j]?

Reply:
Yep, I'd agree with your pre-edit post - it's vile.  I think my silly premature optimisation-obsessed brain was thinking that checking equality to 1 then setting to -1 or 1 might result in fewer instructions than multiplying by -1, but it's negligible compared to the other stuff we're doing anyway (like generating random numbers) if it's even better at all.  Changed accordingly.